### PR TITLE
Refactor Command Patterns

### PR DESCRIPTION
**Pull request by tanndlin**
Currently, I didn't override `validate` if there were no required args, for the sake of less lines. Lmk if I should include it anyways

### CODE REVIEW

File: src/app/plugins/modreport.plugin.ts [hunk 1]
```diff
@@ -10,22 +10,25 @@ export class ModReportPlugin extends Plugin {
...
+  public _commandPattern: RegExp = /(add|list|warn|ban)\s+([^#]+#\d{4})\s*(.*)/;
 
   constructor(public container: IContainer) {
     super();
   }
 
-  public async execute(message: IMessage, args: string[]) {
-    const match_arr = args.join(' ').match(this._commandPattern);
+  public validate(message: IMessage, args: string[]) {
+    if (!args.length) return false;
```
Comment (CallumCoy) [inline src/app/plugins/modreport.plugin.ts]: why didn't you just do the following?
`
const match_arr = args.join(' ').match(this._commandPattern)
return !!match_arr
`

Comment (CallumCoy) [reply src/app/plugins/modreport.plugin.ts]: Also, is there a reason you overwrite the validate function for this one despite it looking like it does the same thing?

Comment (tanndlin) [reply src/app/plugins/modreport.plugin.ts]: Already changed that in another commit

File: src/app/plugins/slowmode.plugin.ts [hunk 2]
```diff
@@ -10,15 +10,12 @@ export class SlowModePlugin extends Plugin {
   public pluginAlias = ['slow'];
   public permission: ChannelType = ChannelType.Staff;
   public pluginChannelName: string = Constants.Channels.Staff.ModChat;
+  public commandPattern: RegExp = /(\d+\s){2}[a-z]{3}\d{4}_[a-z]+/;
```
Comment (c650) [inline src/app/plugins/slowmode.plugin.ts:13]: this wont always work because i dont think cop3100c_fadfdad will match

**Review by c650 — COMMENTED** [review]
this is an interesting idea but i would separate the commandpattern feature from the use of it in many many plugins... it will be easy to introduce bug. 

instead, make minimal pr for commandPattern and then just as we go along we can refactor older plugins.

one cool thing (but may not add much value), is actually deliver the match groups to the plugin so that we don't have to run the regex twice

**Comment by tanndlin** [comment]
waiting for #337 before continuing

**Comment by tanndlin** [comment]
> one cool thing (but may not add much value), is actually deliver the match groups to the plugin so that we don't have to run the regex twice
I feel like this should be a different PR

**Comment by c650** [comment]
> I feel like this should be a different PR

definitely

**Review by c650 — APPROVED** [review]

**Comment by c650** [comment]
let me run a deploy before we cehck this in

**Comment by c650** [comment]
good to check this in whenever

**Comment by tanndlin** [comment]
tested every plugin I edited in this PR 🤞

**Pull request merged**